app/javascript/lib/study-overview.js

```diff
@@ -117,10 +117,18 @@ function resizePlots() {
   }
 }
 
+/** Set colors for group-based annotations */
+export function setMarkerColors(data) {
+  return data.map((trace, i) => {
```

Comment:
That's cool - I didn't know you could have nested returns in JS to change an object in place like this.